src/params/index.ts

```diff
@@ -97,3 +97,38 @@ export function createBaseValues(values: WritableParams): Params {
   )
   return result as Params
 }
+
+export function modify(name: keyof WritableBaseParams) {
+  return {
+    name,
+    value: 0 as number,
+    percent: 0 as Percent,
+    condition: true as boolean,
+
+    byPercent(percent: Percent) {
+      this.percent = percent
+      return this
+    },
+
+    byValue(value: number) {
+      this.value = value
+      return this
+    },
+
+    if(condition: boolean) {
+      this.condition = condition
+      return this
+    },
+  }
+}
+
+export type Change = ReturnType<typeof modify>
+
+export function applyEffects(values: BaseParams, changes: Change[]): BaseParams {
+  changes
+    .filter((change) => change.condition)
+    .forEach((change) => {
+      values[change.name] += change.value || (values[change.name] * change.percent!) / 100
```

Comment:
Unfortunately, this changes the semantics of the laws: `changePercentBy()`  was there to ensure that a percentage value, such as `popularity` or `carRenewablePercentage` is changed by the absolute percentage value not by a relative percentage.

On the other hand, the functions in `lawTools.ts` ensure that the boundaries of the repsective parameter are ensured. E.g. `changePercentBy()` ensures that the percentage is never below 0% or above 100%.

I will check out the branch and try to correct that.